packages/Toggle/src/Toggle.js

```diff
@@ -424,12 +426,12 @@ export default class Toggle extends PureComponent {
     return (
       <label
         className={cx(statefulStyles.container, className)}
-        htmlFor={this.checkboxId}
+        htmlFor={id || this.checkboxId}
```

Comment:
We missed `const labelId = `${this.checkboxId}-label`;` on line 181.

What if we turn `checkboxId` to a function and just include this branching inside there?

Reply:
I don't see what you are referring to on 181, but is this what you had in mind for `checkboxId`

`checkboxId = this.props.id || `toggle-${Math.floor(Math.random() * 10000000)}`;`

or

```const getCheckboxId = () => { 
  return this.props.id || `toggle-${Math.floor(Math.random() * 10000000)}`;`
}```

Reply:
`const labelId = `${this.checkboxId}-label`;` line. That should use the passed in ID right?

It would have to be a function so it always reflects the latest value.

Reply:
Oh that's on the Checkbox component! I was only updating Toggle... probably makes sense to update both though

Reply:
@hswolff Could you clarify what you mean by "It would have to be a function so it always reflects the latest value."?

Wouldn't simply doing something like this before returning the JSX, and using this value be sufficient?
```js
const {id} = this.props
const checkboxId = id || this.checkboxId
```

Reply:
Having one location to get an ID prevents the whackamole of finding every usage of `this.checkbox` and prefixing it with `this.props.id`. It's not as much of an issue for Toggle, but is more so with Checkbox that has more usages of it.

Reply:
Even though checkbox is still a class, we don't use the IDs outside of the render method, and a const should be just fine in that case.

Reply:
@hswolff we fixed so that it's defined in only one location